src/ui/molecules/header/Header.tsx

```diff
@@ -1,30 +1,106 @@
-import React from 'react'
-import './header.scss'
-import { ThemeSwitcher } from 'ui/atoms/theme/ThemeSwitcher'
-import type { DeepReadonly } from 'superTypes'
-import { useBreakpoint } from 'hook/useBreakpoint'
-import type { Breakpoints } from 'hook/useBreakpoint'
+import React, { useCallback, useState } from 'react'
+import type { DeepReadonly, UseState } from 'superTypes'
 import classNames from 'classnames'
+import { useMediaType } from 'hook/useMediaType'
+import { ThemeSwitcher as Switcher } from 'ui/atoms/theme/ThemeSwitcher'
+import { Icon } from 'ui/atoms/icon/Icon'
+import './header.scss'
 
 export type HeaderProps = {
   /**
    * Element positioned first in the reading flow.
    */
   readonly firstElement: JSX.Element
+  /**
+   * The list of navigable links that make up the menu
+   */
+  readonly navigationMenu?: JSX.Element[]
+}
+
+type BurgerMenuProps = {
+  openedMenu: boolean
+  handleBurgerMenuOpen: () => void
+}
+
+type BurgerMenuListProps = {
+  readonly navigationMenu: JSX.Element[]
 }
 
+const BurgerMenu = ({
+  handleBurgerMenuOpen,
+  openedMenu
+}: DeepReadonly<BurgerMenuProps>): JSX.Element => (
+  <div
+    className={classNames(
+      'okp4-header-navigation-burger-menu',
+      openedMenu ? 'rotate-up' : 'rotate-down'
+    )}
+    onClick={handleBurgerMenuOpen}
+  >
+    <Icon name="menu" />
+  </div>
+)
+
+const BurgerMenuList = ({ navigationMenu }: DeepReadonly<BurgerMenuListProps>): JSX.Element => (
+  <div className="okp4-header-navigation-menu-list">
+    {navigationMenu.map((link: Readonly<JSX.Element>, index: number) => (
+      <div className="okp4-header-navigation-menu-item" key={index}>
+        {link}
+      </div>
+    ))}
+  </div>
+)
+
+const ThemeSwitcher = (): JSX.Element => (
+  <div className="okp4-header-theme-switcher-container">
+    <Switcher />
+  </div>
+)
+
 export const Header: React.FC<HeaderProps> = ({
-  firstElement
+  firstElement,
+  navigationMenu
 }: DeepReadonly<HeaderProps>): JSX.Element => {
-  const { isXSmall }: Breakpoints = useBreakpoint()
-  const containerClass = classNames('okp4-header-first-element-container', isXSmall && 'x-small')
+  const [open, setOpen]: UseState<boolean> = useState<boolean>(false)
+  const displayBurgerMenu = useMediaType('(max-width: 995px)')
+  const showBurgerList = displayBurgerMenu && open
+  const headerClassname = classNames(
+    'okp4-header-main',
+    navigationMenu ? 'with-navigation' : 'without-navigation',
+    {
+      'burger-list': showBurgerList
+    }
+  )
+  const handleBurgerMenuOpen = useCallback((): void => {
+    setOpen(!open)
+  }, [open, setOpen])
 
   return (
```

Comment:
In the return there are nested logic expressions. How about we move these to the declared components above instead to facilitate readability? We'd end up with something more to the like of the `Toast` component.

Reply:
I refactored to make it clearer but I'm not in favor of creating a component that returns `null` if a condition is not met.
I prefer to have a condition that returns or not the component.
I find it unclear to have a component , for example `<Button / >`, in the JSX and in the `Button` code we find out that it can return null.